### PR TITLE
feat(plasma-b2c, plasma-web): autoResize for TextArea

### PR DESCRIPTION
**Pull request by kayman233**
Добавлена возможность автоматического регулирования высоты поля ввода по длине контента внутри (параметра `value`) с помощью свойства `autoResize`. 
В этом режиме можно указать крайние значения высоты поля ввода, используя свойства `minAuto`, `maxAuto`,
указав их в `rem`.

```tsx live
const [value, setValue] = React.useState('Значение');

return (
    <TextArea
        placeholder="Введите значение"
        value={value}
        onChange={(e) => {
            setValue(e.target.value);
        }}
        autoResize
        minAuto={3}
        maxAuto={5}
    />
);
```
ezgif-5-f069f315b1

<details>
  <summary>:baby_chick: Download canary assets:</summary>
  <blockquote>
    <details>
      <summary>color</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_ios-swift--canary.614.5553174292.swift'>color_asdk_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_kotlin--canary.614.5553174292.kt'>color_asdk_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_react-native--canary.614.5553174292.ts'>color_asdk_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_xml--canary.614.5553174292.xml'>color_asdk_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_ios-swift--canary.614.5553174292.swift'>color_plasma_b2c_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_kotlin--canary.614.5553174292.kt'>color_plasma_b2c_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_react-native--canary.614.5553174292.ts'>color_plasma_b2c_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_xml--canary.614.5553174292.xml'>color_plasma_b2c_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.614.5553174292.swift'>color_sberHealth_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.614.5553174292.kt'>color_sberHealth_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.614.5553174292.ts'>color_sberHealth_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_xml--canary.614.5553174292.xml'>color_sberHealth_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.614.5553174292.swift'>color_sbermarket_business_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.614.5553174292.kt'>color_sbermarket_business_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.614.5553174292.ts'>color_sbermarket_business_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_xml--canary.614.5553174292.xml'>color_sbermarket_business_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.614.5553174292.swift'>color_sbermarket_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.614.5553174292.kt'>color_sbermarket_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.614.5553174292.swift'>color_sbermarket_metro_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.614.5553174292.kt'>color_sbermarket_metro_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.614.5553174292.ts'>color_sbermarket_metro_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_xml--canary.614.5553174292.xml'>color_sbermarket_metro_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.614.5553174292.ts'>color_sbermarket_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.614.5553174292.swift'>color_sbermarket_selgros_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.614.5553174292.kt'>color_sbermarket_selgros_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.614.5553174292.ts'>color_sbermarket_selgros_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_xml--canary.614.5553174292.xml'>color_sbermarket_selgros_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.614.5553174292.swift'>color_sbermarket_wlbusiness_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.614.5553174292.kt'>color_sbermarket_wlbusiness_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.614.5553174292.ts'>color_sbermarket_wlbusiness_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_xml--canary.614.5553174292.xml'>color_sbermarket_wlbusiness_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_xml--canary.614.5553174292.xml'>color_sbermarket_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.614.5553174292.swift'>color_sberonline_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.614.5553174292.kt'>color_sberonline_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.614.5553174292.ts'>color_sberonline_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_xml--canary.614.5553174292.xml'>color_sberonline_xml--canary.614.5553174292.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.614.5553174292.swift'>color_sberprime_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.614.5553174292.kt'>color_sberprime_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.614.5553174292.ts'>color_sberprime_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_xml--canary.614.5553174292.xml'>color_sberprime_xml--canary.614.5553174292.xml</a><br>
      </blockquote>
    </details>
    <details>
      <summary>shadow</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.614.5553174292.ts'>shadow_sbermarket_react-native--canary.614.5553174292.ts</a><br>
      </blockquote>
    </details>
    <details>
      <summary>typo</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.614.5553174292.swift'>typo_mage_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.614.5553174292.kt'>typo_mage_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.614.5553174292.ts'>typo_mage_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.614.5553174292.swift'>typo_plasma_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.614.5553174292.kt'>typo_plasma_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.614.5553174292.ts'>typo_plasma_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.614.5553174292.swift'>typo_ruler_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.614.5553174292.kt'>typo_ruler_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.614.5553174292.ts'>typo_ruler_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.614.5553174292.swift'>typo_sage_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.614.5553174292.kt'>typo_sage_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.614.5553174292.ts'>typo_sage_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.614.5553174292.swift'>typo_sbermarket_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.614.5553174292.kt'>typo_sbermarket_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.614.5553174292.ts'>typo_sbermarket_react-native--canary.614.5553174292.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.614.5553174292.swift'>typo_soulmate_ios-swift--canary.614.5553174292.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.614.5553174292.kt'>typo_soulmate_kotlin--canary.614.5553174292.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.614.5553174292.ts'>typo_soulmate_react-native--canary.614.5553174292.ts</a><br>
      </blockquote>
    </details>
  </blockquote>
</details>

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.219.1-canary.614.5553174292.0
  npm install @salutejs/plasma-hope@1.219.1-canary.614.5553174292.0
  npm install @salutejs/plasma-web@1.219.1-canary.614.5553174292.0
  # or 
  yarn add @salutejs/plasma-b2c@1.219.1-canary.614.5553174292.0
  yarn add @salutejs/plasma-hope@1.219.1-canary.614.5553174292.0
  yarn add @salutejs/plasma-web@1.219.1-canary.614.5553174292.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-614/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-614/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-614/web-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-614/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-614/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-614/web-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-614/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-614/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-614/web-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-614/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-614/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-614/web-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-614/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-614/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-614/web-storybook/

**Comment by Yakutoc** [comment]
@kayman233 Добавь пожалуйста короткое описание для PR?

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-614/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-614/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-614/web-storybook/

File: packages/plasma-hope/src/components/TextArea/useAutoResize.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { useEffect, useRef } from 'react';
+
+const ROOT_FONT_SIZE = 16;
```
Comment (neretin-trike) [inline packages/plasma-hope/src/components/TextArea/useAutoResize.ts:3]: предлагаю экспортнуть эту переменную из TextArea.tsx, чтобы не следить за этими переменными в двух местах

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

File: packages/plasma-hope/src/components/TextArea/views/web/TextAreaWeb.tsx [hunk 2]
```diff
@@ -46,8 +48,17 @@ export const TextAreaWeb = ({
...
+    value,
+    autoResize = false,
+    minAuto = 0,
+    maxAuto,
     ...rest
 }: TextAreaPropsCommon) => {
+    const inputRef = useRef<HTMLTextAreaElement | null>(null);
+    const forkRef = useForkRef(inputRef, outerRef);
+
+    useAutoResize(autoResize, inputRef, value, minAuto, maxAuto);
```
Comment (neretin-trike) [inline packages/plasma-hope/src/components/TextArea/views/web/TextAreaWeb.tsx:60]: а что думаешь на счёт выноса этого хука на уровень выше в TextArea.tsx с прокидыванием всех свойству сюда? Чтобы не дублировать эту логику в оба View, т.к. функционал по сути тот же самый

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-614/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-614/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-614/web-storybook/

File: packages/plasma-hope/src/components/TextArea/views/web/TextAreaWeb.tsx (hunk 2 above)
Comment (kayman233) [reply packages/plasma-hope/src/components/TextArea/views/web/TextAreaWeb.tsx]: да, ты прав. Теперь в одном месте всё

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-614/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-614/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-614/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-614/web-storybook/

**Review by Yeti-or — APPROVED** [review]

**Comment by Salute-Eva** [comment]
🚀 This PR is included in version: @salutejs/plasma-b2c@1.220.0, @salutejs/plasma-core@1.125.0, @salutejs/plasma-cy-utils@0.65.0, @salutejs/plasma-hope@1.220.0, @salutejs/plasma-icons@1.154.0, @salutejs/plasma-sb-utils@0.123.0, @salutejs/plasma-temple-docs@0.160.0, @salutejs/plasma-temple@1.169.0, @salutejs/plasma-tokens-b2b@1.24.0, @salutejs/plasma-tokens-b2c@0.33.0, @salutejs/plasma-tokens-native@1.9.0, @salutejs/plasma-tokens-utils@0.29.0, @salutejs/plasma-tokens-web@1.39.0, @salutejs/plasma-tokens@1.58.0, @salutejs/plasma-typo@0.36.0, @salutejs/plasma-ui-docs@0.209.0, @salutejs/plasma-ui@1.201.0, @salutejs/plasma-web-docs@0.178.0, @salutejs/plasma-web@1.220.0, @salutejs/plasma-website@0.186.0 🚀

**Pull request merged**